src/main/java/org/broadinstitute/dsm/route/MercuryDummyKitEndpoint.java

```diff
@@ -0,0 +1,132 @@
+        logger.info("Found kit type for Mercury Dummy Endpoint " + kitTypeId);
+        
+        //if instance not null
+        writeRequest(mockDdpInstance.getDdpInstanceId(), mercuryKitRequestId, kitTypeId,
+                ddpParticipantId, participantCollaboratorId, collaboratorSampleId,
+                "SYSTEM", kitLabel);
+
+        return new Result(200);
+    }
+
+    static String writeRequest(@NonNull String instanceId, @NonNull String ddpKitRequestId, @NonNull int kitTypeId,
+                               @NonNull String ddpParticipantId, String collaboratorPatientId, String collaboratorSampleId,
+                               @NonNull String createdBy, String kitLabel) {
+        SimpleResult results = inTransaction((conn) -> {
+            SimpleResult dbVals = new SimpleResult(0);
+            try (PreparedStatement insertKitRequest = conn.prepareStatement(TransactionWrapper.getSqlFromConfig(ApplicationConfigConstants.INSERT_KIT_REQUEST), Statement.RETURN_GENERATED_KEYS)) {
+                insertKitRequest.setString(1, instanceId);
+                insertKitRequest.setString(2, ddpKitRequestId);
+                insertKitRequest.setInt(3, kitTypeId);
+                insertKitRequest.setString(4, ddpParticipantId);
+                insertKitRequest.setObject(5, collaboratorPatientId);
+                insertKitRequest.setObject(6, collaboratorSampleId);
+                insertKitRequest.setNull(7, Types.VARCHAR); //ddp_label or shipping_id
+                insertKitRequest.setString(8, createdBy);
+                insertKitRequest.setLong(9, System.currentTimeMillis());
+                insertKitRequest.setNull(10, Types.VARCHAR); //external_order_number
+                insertKitRequest.setNull(11, Types.VARCHAR); //upload reason
+                insertKitRequest.executeUpdate();
+                try (ResultSet rs = insertKitRequest.getGeneratedKeys()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getString(1);
+                    }
+                }
+                catch (Exception e) {
+                    throw new RuntimeException("Error getting id of new kit request ", e);
+                }
+            }
+            catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            if (dbVals.resultException == null && dbVals.resultValue != null) {
+                writeNewKit(conn, (String) dbVals.resultValue, kitLabel);
+            }
+            return dbVals;
+        });
+
+        if (results.resultException != null) {
+            throw new RuntimeException("Error adding kit request  w/ ddpKitRequestId " + ddpKitRequestId, results.resultException);
+        }
+
+        logger.info("Added kitRequest w/ ddpKitRequestId " + ddpKitRequestId);
+        return (String) results.resultValue;
+    }
+
+    private static SimpleResult writeNewKit(Connection conn, String kitRequestId, String kiLabel) {
```

Comment:
if you can use the `KitRequestShipping.writeRequest` you could then add the label per `KitStatusChangeRoute.updateKit()`